### DISCUSSION TITLE
Function returning json

### DISCUSSION THREAD

**Discussion by gmijo47** [nlohmann/json#3052] [Q&A] [answered]
Hi there, just wanna know is it possible for the function to return json  like down below
In function.h
```
json i = json::object();
		i["filever"]["application"][0] = { {"version"}, version };
		return i;
```
In client.cpp
```
 json v = json::object();
            v = A_Data::getVersion();
```
And im getting this error [json.exception.type_error.302] type must be string, but is array

**Comment by nlohmann** [comment]
This should work. With the code you provided, I cannot diagnose further. The error message could mean that the key `version` is not a `std::wstring` (which will not be interpreted as string, but as array of chars). But without a working example, I can only guess.

**Reply by gmijo47** [reply]
@nlohmann , here is full example.
**Main.cpp**
```
#include <atomic>
#include <iostream>
#include <string>
#include "../json/json.hpp"
#include "fx.h"
using json = nlohmann::json;

int main() {
	std::string cmd = "";
	std:: cin >> cmd;
	if (cmd == "getVer"){

		json v = json::object();
		v = getVersion();
		std::cout << v.dump();
	}
}
```
**fx.h - stands for function header**
```
#include "../json/json.hpp"
using json = nlohmann::json;
#include <string>

inline std::string getVersion() {
	std::string version = "3.3.3";
	json i = json::object();
	i["filever"]["application"][0] = { {"version"}, version };
	return i;
}
```
Also, i tried to use different compilers such as MSVC, g++/gcc...

**Reply by gregmarr** [reply]
inline std::string getVersion() {
        json i;
        return i;
    }

Your function is declared as returning `std::string` but you're returning a json object.

**Comment by gmijo47** [comment, marked as answer]
@gregmarr thanks, that was a problem. U are awesome man 😄

**Reply by gregmarr** [reply]
See this section for how to make the compiler catch this for you: https://github.com/nlohmann/json#implicit-conversions
